src/com/lftechnology/java/training/sanish/stemleaf/StemLeafDemo.java

```diff
@@ -0,0 +1,89 @@
+package com.lftechnology.java.training.sanish.stemleaf;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class StemLeafDemo {
+	private static final Logger LOGGER = Logger.getLogger(StemLeafDemo.class
+			.getName());
+	private static ArrayList<Integer> numberPoints = new ArrayList<Integer>();
+	private static ArrayList<StemLeaf> stemLeafs = new ArrayList<StemLeaf>();
+
+	public static void main(String[] args) {
+		try (Scanner scanner = new Scanner(System.in)) {
+			getDataPoints(scanner);
+			generateStemLeafs();
+			displayStemLepfs();
+		} catch (Exception e) {
+			LOGGER.log(Level.WARNING, "Exception : {0}",
+					new Object[] { e.getMessage() });
+		}
+	}
+
+	/**
+	 * Get data points for console
+	 * @param scanner {@link Scanner}
+	 * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+	 */
+	public static void getDataPoints(Scanner scanner) {
+		LOGGER.log(Level.INFO, "Enter number of data points :");
+		int dataPoints = UserInput.getNumber(scanner, Constants.NUM_ZERO, Constants.MAX_NUM);
+		int userInputNum;
+		for (int i = 0; i < dataPoints; i++) {
+			LOGGER.log(Level.INFO, "[{0}]", new Object[] { i });
+			userInputNum = UserInput.getNumber(scanner, Constants.MIN_NUM,
+					Constants.MAX_NUM);
+			numberPoints.add(userInputNum);
+		}
+	}
+
+	/**
+	 * Generate stem & leafs form data points
+	 * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+	 */
+	public static void generateStemLeafs() {
+		int stem, leaf, firstNum, secondNum, leafCount;
+		int dataPointLength = numberPoints.size();
+		StemLeaf stemLeaf;
+		while (dataPointLength != 0) {
+			stem = numberPoints.get(0) / 10;
+			leafCount = 0;
+			leaf = 0;
+			for (int j = 0; j < dataPointLength; j++) {
+				firstNum = numberPoints.get(j) / 10;
+				if (firstNum == stem) {
+					leafCount++;
+					secondNum = numberPoints.get(j) % 10;
+					leaf = leaf * 10 + secondNum;
+					numberPoints.remove(j);
+					dataPointLength--;
+					j--;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor the code in order to not assign to this loop counter from within the loop body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AForLoopCounterChangedCheck)